build/patches/Always-ask-for-permissions.patch

```diff
@@ -0,0 +1,291 @@
+--- a/components/page_info/android/java/src/org/chromium/components/page_info/PermissionParamsListBuilder.java
++++ b/components/page_info/android/java/src/org/chromium/components/page_info/PermissionParamsListBuilder.java
+@@ -85,6 +85,10 @@ public class PermissionParamsListBuilder {
+                     permissionParams.warningTextResource =
+                             R.string.page_info_android_permission_blocked;
+                 }
++                if (AndroidPermissionRequester.alreadyAskedPermissionToUser(permission.type) == false) {
++                    permissionParams.warningTextResource =
++                            R.string.page_info_android_permission_user_permission_required;
++                }
+             }
+         }
+ 
+diff --git a/components/permissions/android/java/src/org/chromium/components/permissions/AndroidPermissionRequester.java b/components/permissions/android/java/src/org/chromium/components/permissions/AndroidPermissionRequester.java
+--- a/components/permissions/android/java/src/org/chromium/components/permissions/AndroidPermissionRequester.java
++++ b/components/permissions/android/java/src/org/chromium/components/permissions/AndroidPermissionRequester.java
+@@ -26,6 +26,10 @@ import org.chromium.ui.modelutil.PropertyModel;
+ 
+ import java.util.HashSet;
+ import java.util.Set;
++import java.util.Hashtable;
++import java.time.LocalDateTime;
++import java.time.temporal.ChronoUnit;
++import org.chromium.base.ContextUtils;
+ 
+ /**
+  * Methods to handle requesting native permissions from Android when the user grants a website a
+@@ -79,6 +83,9 @@ public class AndroidPermissionRequester {
+     public static boolean hasRequiredAndroidPermissionsForContentSetting(
+             AndroidPermissionDelegate permissionDelegate,
+             @ContentSettingsType int contentSettingsType) {
++        if (alreadyAskedPermissionToUser(contentSettingsType) == false)
++            return false;
++
+         Set<String> missingPermissions = filterPermissionsKeepMissing(permissionDelegate,
+                 PermissionUtil.getRequiredAndroidPermissionsForContentSetting(contentSettingsType));
+ 
+@@ -148,6 +155,8 @@ public class AndroidPermissionRequester {
+                         if (!windowAndroid.canRequestPermission(permissions[i])) {
+                             allRequestable = false;
+                         }
++                    } else if (grantResults[i] == PackageManager.PERMISSION_GRANTED) {
++                        setUserGrantedPermission(permissions[i]);
+                     }
+                 }
+ 
+@@ -198,6 +207,13 @@ public class AndroidPermissionRequester {
+         return true;
+     }
+ 
++    public static void showMissingPermissionDialogUnchecked(Activity activity, @StringRes int messageId,
++            Runnable onPositiveButtonClicked, Runnable onCancelled) {
++        // permitted without any check for requestFileAccessPermissionHelper.FileAccessPermissionHelper
```

Comment:
Where is the filter for that specific `FileAccessPermissionHelper` applied?

Reply:
I do not understand.

that function is for the download manager to request access to the storage, I excluded it from the checks.

Reply:
Ok; so I think the comment should be something like:
```
// do not re-request permissions for file access
```

Reply:
fixed in https://github.com/bromite/bromite/pull/1606/commits/e5a236efbc5c2fa74667089fa3d4c508563fc4aa